examples/server/server.cpp

```diff
@@ -1291,9 +1303,19 @@ struct server_queue {
                         QUE_DBG("%s", "terminate\n");
                         return;
                     }
-                    condition_tasks.wait(lock, [&]{
-                        return (!queue_tasks.empty() || !running);
-                    });
+                    const auto pred = [&] {
+                            return (!queue_tasks.empty() || !running);
+                    };
+                    if (standby_timeout > 0) {
+                        if (!condition_tasks.wait_for(lock, std::chrono::seconds(standby_timeout), pred)) {
+                            lock.release()->unlock(); // unlock the unique_lock, before calling the shutdown_handler, as it tries to lock it
+                            QUE_INF("%s", "stand-by timeout reached\n");
+                            shutdown_handler(::standby_timeout{});
```

Comment:
You're abusing `shutdown_handler` here. It is supposed to be called by signal handler. If you ever want to stop server_queue, simply set `running = false` then break. No need fancy calls here.